astropy/units/core.py

```diff
@@ -1480,7 +1480,7 @@ def __init__(self, st, doc=None, format=None, namespace=None):
         elif isinstance(st, tuple):
             if not len(st) == 2:
                 raise ValueError("st must be string, list or 2-tuple")
-            self._names = st[0] + st[1]
+            self._names = st[0] + [n for n in st[1] if n not in st[0]]
```

Comment:
You may lose the special unique identifier (first element of the short names) if you convert it to a ``set``.

But there is a recipe for a ``unique_everseen`` function in the [Python itertools documentation](https://docs.python.org/2/library/itertools.html#recipes) (needs some scrolling)